pydantic/_internal/_core_utils.py

```diff
@@ -83,7 +83,7 @@ def get_type_ref(type_: type[Any], args_override: tuple[type[Any], ...] | None =
     if isinstance(origin, TypeAliasType):
         type_ref = f'{module_name}.{origin.__name__}'
     else:
-        qualname = getattr(origin, '__qualname__', f'<No __qualname__: {origin}>')
+        qualname = getattr(origin, '__qualname__', '<No __qualname__>')
```

Comment:
We can have a fallback to `__str__` if wanted... 🤷‍♂️

Reply:
The thing is that `repr()` is always called, even if `origin.__qualname__` exists... 🤔

Anyway... This PR solves the issue.